### ISSUE TITLE
Getting error after upgrading to 4.

### ISSUE THREAD

**Issue by jarrettj** [Codeception/Codeception#5860]
#### What are you trying to achieve?
Upgrade to 4.1.1
#### What do you get instead?
An error.
> Provide console output if related. Use `-vvv` mode for more details.

```bash
$ php vendor/bin/codecept run --env docker-local-chrome --fail-fast -vvv
Codeception PHP Testing Framework v4.1.1
Powered by PHPUnit 5.7.27 by Sebastian Bergmann and contributors.
Running with seed: 

PHP Fatal error:  Interface 'Symfony\Contracts\Service\ResetInterface' not found in /var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcherInterface.php on line 23
```
> Provide test source code if related

```php
Doesn't make it to run a test.
```
### Details

* Codeception version:4.1.1
* PHP Version:7.1
* Operating System:Ubuntu
* Installation type:Composer
* List of installed packages (`composer show`) See below
* Suite configuration:
```
class_name: WebGuy
modules:
    enabled:
        - WebDriver
        - Db
        - Redis
config:
        WebDriver:
            url: 'https://localhost/'
            browser: firefox
            port: 4444
        Redis:
            host: 'localhost'
            port: 6379
            database: 0
        Db:
            dsn: 'mysql:host=localhost;dbname=ci_test_legal'
            user: 'root'
            password: 'root'
            dump: 'module/Application/tests/_data/dump.sql'
            populate: true
            cleanup: false
env:
    docker-local-chrome-shared:
        modules:
            config:
                WebDriver:
                    url: 'https://legal_testing_app'
                    browser: chrome
                    host: legal_selenium_hub
                    clear_cookies: false
                    restart: false
                    wait: 10
                Db:
                    dsn: 'mysql:host=legal_testing_mysql;dbname=ci_test_legal'
                    user: 'root'
                    password: 'root'
                    dump: 'module/Application/tests/_data/dump.sql'
                    populate: false
                    cleanup: false
                Redis:
                    host: 'legal_testing_redis'
                    port: 6379
                    database: 0

```

```
$ composer show
aws/aws-sdk-php                     3.121.1            AWS SDK for PHP - Use Amazon Web Services in your PHP project
aws/aws-sdk-php-zf2                 3.0.0              Zend Framework 2 Module that allows easy integration the AWS SDK for PHP
beberlei/doctrineextensions         v1.2.2             A set of extensions to Doctrine 2 that add support for additional query functions available in MySQL and Oracle.
behat/gherkin                       v4.6.0             Gherkin DSL parser for PHP 5.3
behat/transliterator                v1.2.0             String transliterator
bjyoungblood/bjy-authorize          1.5.0-beta1        Zend\Acl based firewall system for ZF2 dispatch protection
bjyoungblood/BjyProfiler            dev-master 1d05f9b Database profiler for Zend\Db (also plugin for ZendDeveloperTools)
codeception/codeception             4.1.1              BDD-style testing framework
codeception/lib-asserts             1.2.0              Assertion methods used by Codeception core and Asserts module
codeception/lib-innerbrowser        1.3.1              Parent library for all Codeception framework modules and PhpBrowser
codeception/module-asserts          1.1.1              Codeception module containing various assertions
codeception/module-db               1.0.1              WebDriver module for Codeception
codeception/module-redis            1.0.0              Redis module for Codeception
codeception/module-webdriver        1.0.3              WebDriver module for Codeception
codeception/module-zf2              1.0.3              Codeception module for Zend Framework 2 and 3
codeception/phpunit-wrapper         6.0.17             PHPUnit classes used by Codeception
codeception/robo-paracept           dev-master ff00b07 Codeception Parallel Execution Tasks via Robo Task Runner
codeception/stub                    2.1.0              Flexible Stub wrapper for PHPUnit's Mock Builder
consolidation/annotated-command     2.12.0             Initialize Symfony Console commands from annotated command class methods.
consolidation/config                1.2.1              Provide configuration services for a commandline tool.
consolidation/log                   1.1.1              Improved Psr-3 / Psr\Log logger based on Symfony Console components.
consolidation/output-formatters     3.5.0              Format text by applying transformations provided by plug-in formatters.
consolidation/robo                  1.4.11             Modern task runner
consolidation/self-update           1.1.5              Provides a self:update command for Symfony Console applications.
container-interop/container-interop 1.2.0              Promoting the interoperability of container objects (DIC, SL, etc.)
dflydev/dot-access-data             v1.1.0             Given a deep data structure, access data by dot notation.
doctrine/annotations                v1.8.0             Docblock Annotations Parser
doctrine/cache                      1.9.1              PHP Doctrine Cache library is a popular cache implementation that supports many different drivers such as redis, m...
doctrine/collections                1.6.4              PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/common                     v2.8.1             Common Library for Doctrine projects
doctrine/dbal                       v2.6.3             Database Abstraction Layer
doctrine/doctrine-module            1.2.0              Zend Framework Module that provides Doctrine basic functionality required for ORM and ODM modules
doctrine/doctrine-orm-module        1.1.5              Zend Framework Module that provides Doctrine ORM functionality
doctrine/inflector                  1.3.1              Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator               1.3.0              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                      1.0.2              PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/migrations                 v1.5.0             Database Schema migrations using Doctrine DBAL
doctrine/orm                        v2.5.14            Object-Relational-Mapper for PHP
facebook/webdriver                  1.7.1              A PHP client for Selenium WebDriver
gedmo/doctrine-extensions           v2.4.37            Doctrine2 behavioral extensions
google/recaptcha                    1.2.3              Client library for reCAPTCHA, a free service that protects websites from spam and abuse.
grasmash/expander                   1.0.0              Expands internal property references in PHP arrays file.
grasmash/yaml-expander              1.4.0              Expands internal property references in a yaml file.
guzzlehttp/guzzle                   6.4.1              Guzzle is a PHP HTTP client library
guzzlehttp/promises                 v1.3.1             Guzzle promises library
guzzlehttp/psr7                     1.6.1              PSR-7 message implementation that also provides common utility methods
hoa/consistency                     1.17.05.02         The Hoa\Consistency library.
hoa/console                         3.17.05.02         The Hoa\Console library.
hoa/event                           1.17.01.13         The Hoa\Event library.
hoa/exception                       1.17.01.16         The Hoa\Exception library.
hoa/file                            1.17.07.11         The Hoa\File library.
hoa/iterator                        2.17.01.10         The Hoa\Iterator library.
hoa/protocol                        1.17.01.14         The Hoa\Protocol library.
hoa/stream                          1.17.02.21         The Hoa\Stream library.
hoa/ustring                         4.17.01.16         The Hoa\Ustring library.
kriswallsmith/assetic               v1.4.0             Asset Management for PHP
league/container                    2.4.1              A fast and intuitive dependency injection container.
linkorb/jsmin-php                   dev-master c598ed5 Unofficial package of jsmin-php
mtdowling/jmespath.php              2.4.0              Declaratively specify how to extract elements from a JSON document
myclabs/deep-copy                   1.9.3              Create deep copies (clones) of your objects
natxet/CssMin                       v3.0.6             Minifying CSS
neeckeloo/newrelic                  v1.2.0             NewRelic for ZF2
neilime/zf2-twb-bundle              2.6.1              Zend Framework 2 module for easy integration of Twitter Bootstrap
ocramius/ocra-service-manager       2.1.2              ZF2 Service manager extensions to allow service proxies and dependency graph generation
ocramius/proxy-manager              1.0.2              A library providing utilities to generate, instantiate and generally operate with Object Proxies
onelogin/php-saml                   3.4.0              OneLogin PHP SAML Toolkit
pclzip/pclzip                       2.8.2              A PHP library that offers compression and extraction functions for Zip formatted archives
php-webdriver/webdriver             1.8.1              A PHP client for Selenium WebDriver. Previously facebook/webdriver.
phpdocumentor/reflection-common     2.0.0              Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock   4.3.2              With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information...
phpdocumentor/type-resolver         1.0.1              A PSR-5 based resolver of Class names, Types and Structural Element Names
phpoffice/common                    0.2.9              PHPOffice Common
phpoffice/phppresentation           0.9.0              PHPPresentation - Read, Create and Write Presentations documents in PHP
phpoffice/phpword                   0.16.0             PHPWord - A pure PHP library for reading and writing word processing documents (OOXML, ODF, RTF, HTML, PDF)
phpspec/prophecy                    1.9.0              Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage           4.0.8              Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator           1.4.5              FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template           1.2.1              Simple template engine.
phpunit/php-timer                   1.0.9              Utility class for timing
phpunit/php-token-stream            2.0.2              Wrapper around PHP's tokenizer extension.
phpunit/phpunit                     5.7.27             The PHP Unit Testing framework.
phpunit/phpunit-mock-objects        3.4.4              Mock Object library for PHPUnit
predis/predis                       v1.1.1             Flexible and feature-complete Redis client for PHP and HHVM
psr/cache                           1.0.1              Common interface for caching libraries
psr/container                       1.0.0              Common Container Interface (PHP FIG PSR-11)
psr/http-message                    1.0.1              Common interface for HTTP messages
psr/log                             1.1.2              Common interface for logging libraries
psr/simple-cache                    1.0.1              Common interfaces for simple caching
ralouphie/getallheaders             3.0.3              A polyfill for getallheaders.
roave/security-advisories           dev-master 40fb2c2 Prevents installation of composer packages with known security vulnerabilities: no API, simply require it
robrichards/xmlseclibs              3.0.4              A PHP library for XML Security
sebastian/code-unit-reverse-lookup  1.0.1              Looks up which function or method a line of code belongs to
sebastian/comparator                1.2.4              Provides the functionality to compare PHP values for equality
sebastian/diff                      1.4.3              Diff implementation
sebastian/environment               2.0.0              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                  2.0.0              Provides the functionality to export PHP variables for visualization
sebastian/global-state              1.1.1              Snapshotting of global state
sebastian/object-enumerator         2.0.1              Traverses array structures and object graphs to enumerate all referenced objects
sebastian/recursion-context         2.0.0              Provides functionality to recursively process PHP variables
sebastian/resource-operations       1.0.0              Provides a list of PHP built-in functions that operate on resources
sebastian/version                   1.0.6              Library that helps with managing the version number of Git-hosted PHP projects
symfony/browser-kit                 v4.4.0             Symfony BrowserKit Component
symfony/console                     v3.4.35            Symfony Console Component
symfony/css-selector                v4.4.0             Symfony CssSelector Component
symfony/debug                       v4.4.0             Symfony Debug Component
symfony/dom-crawler                 v4.4.0             Symfony DomCrawler Component
symfony/event-dispatcher            v4.4.0             Symfony EventDispatcher Component
symfony/event-dispatcher-contracts  v1.1.7             Generic abstractions related to dispatching event
symfony/filesystem                  v4.4.0             Symfony Filesystem Component
symfony/finder                      v4.4.0             Symfony Finder Component
symfony/polyfill-ctype              v1.12.0            Symfony polyfill for ctype functions
symfony/polyfill-mbstring           v1.12.0            Symfony polyfill for the Mbstring extension
symfony/process                     v3.4.35            Symfony Process Component
symfony/yaml                        v3.4.35            Symfony Yaml Component
true/punycode                       v2.1.1             A Bootstring encoding of Unicode for Internationalized Domain Names in Applications (IDNA)
webmozart/assert                    1.6.0              Assertions to validate method input/output with nice error messages.
widmogrod/php-exceptions            0.1.0              Collection PHP exceptions grouped by functionality with nice ::assert API
widmogrod/php-functional            1.4.0              Functors, Applicative and Monads are fascinating concept. Purpose of this library is to explore them in OOP PHP wo...
widmogrod/zf2-assetic-module        1.6.0              Zend Framework 2 module that provides complete integration of Assetic library
zendframework/zend-authentication   2.6.0              provides an API for authentication and includes concrete authentication adapters for common use case scenarios
zendframework/zend-barcode          2.7.1              Programmatically create and render barcodes as images or in PDFs
zendframework/zend-cache            2.8.3              Caching implementation with a variety of storage options, as well as codified caching strategies for callbacks, cl...
zendframework/zend-captcha          2.8.0              Generate and validate CAPTCHAs using Figlets, images, ReCaptcha, and more
zendframework/zend-code             2.6.3              provides facilities to generate arbitrary code using an object oriented interface
zendframework/zend-config           2.6.0              provides a nested object property based user interface for accessing this configuration data within application code
zendframework/zend-console          2.7.0              Build console applications using getopt syntax or routing, complete with prompts
zendframework/zend-crypt            2.6.0             
zendframework/zend-db               2.10.0             Database abstraction layer, SQL abstraction, result set abstraction, and RowDataGateway and TableDataGateway imple...
zendframework/zend-debug            2.6.0              Safely dump debug information to HTML
zendframework/zend-developer-tools  1.2.3              Module for developer and debug tools for use with zend-mvc applications.
zendframework/zend-di               2.6.1             
zendframework/zend-diactoros        1.8.7              PSR HTTP Message implementations
zendframework/zend-dom              2.7.2              provides tools for working with DOM documents and structures
zendframework/zend-escaper          2.6.1              Securely and safely escape HTML, HTML attributes, JavaScript, CSS, and URLs
zendframework/zend-eventmanager     2.6.4             
zendframework/zend-feed             2.10.3             provides functionality for consuming RSS and Atom feeds
zendframework/zend-file             2.8.3              Locate PHP classfiles
zendframework/zend-filter           2.9.2              Programmatically filter and normalize data and files
zendframework/zend-form             2.13.0             Validate and display simple and complex forms, casting forms to business objects and vice versa
zendframework/zend-http             2.8.4              Provides an easy interface for performing Hyper-Text Transfer Protocol (HTTP) requests
zendframework/zend-hydrator         1.1.0             
zendframework/zend-i18n             2.10.0             Provide translations for your application, and filter and validate internationalized values
zendframework/zend-i18n-resources   2.6.1              Provides validator translations for zend-i18n's Translator
zendframework/zend-inputfilter      2.10.1             Normalize and validate input sets from the web, APIs, the CLI, and more, including files
zendframework/zend-json             2.6.1              provides convenience methods for serializing native PHP to JSON and decoding JSON to native PHP
zendframework/zend-loader           2.6.1              Autoloading and plugin loading strategies
zendframework/zend-log              2.11.0             Robust, composite logger with filtering, formatting, and PSR-3 support
zendframework/zend-mail             2.10.0             Provides generalized functionality to compose and send both text and MIME-compliant multipart e-mail messages
zendframework/zend-math             2.7.1             
zendframework/zend-memory           2.6.1              Manage data in an environment with limited memory
zendframework/zend-mime             2.7.2              Create and parse MIME messages and parts
zendframework/zend-modulemanager    2.8.4              Modular application system for zend-mvc applications
zendframework/zend-mvc              2.7.15            
zendframework/zend-navigation       2.9.1              Manage trees of pointers to web pages in order to build navigation systems
zendframework/zend-paginator        2.8.2              Paginate collections of data from arbitrary sources
zendframework/zend-permissions-acl  2.7.1              Provides a lightweight and flexible access control list (ACL) implementation for privileges management
zendframework/zend-permissions-rbac 2.6.0              provides a role-based access control management
zendframework/zend-progressbar      2.6.0              Create and update progress bars in different environments
zendframework/zend-psr7bridge       0.2.2              PSR-7 <-> Zend\Http bridge
zendframework/zend-serializer       2.9.1              Serialize and deserialize PHP structures to a variety of representations
zendframework/zend-server           2.8.1              Create Reflection-based RPC servers
zendframework/zend-servicemanager   2.7.11            
zendframework/zend-session          2.8.7              manage and preserve session data, a logical complement of cookie data, across multiple page requests by the same c...
zendframework/zend-soap             2.8.0             
zendframework/zend-stdlib           2.7.7             
zendframework/zend-tag              2.7.1              Manipulate and weight taggable items, and create tag clouds
zendframework/zend-test             2.6.1             
zendframework/zend-text             2.7.1              Create FIGlets and text-based tables
zendframework/zend-uri              2.7.1              A component that aids in manipulating and validating » Uniform Resource Identifiers (URIs)
zendframework/zend-validator        2.11.1             provides a set of commonly needed validators
zendframework/zend-version          2.5.1             
zendframework/zend-view             2.11.3             Flexible view layer supporting and providing multiple view layers, helpers, and more
zendframework/zend-xmlrpc           2.7.0              Fully-featured XML-RPC server and client implementations
zendframework/zenddiagnostics       v1.6.0             A set of components for performing diagnostic tests in PHP applications
zendframework/zendframework         2.5.3              Zend Framework 2
zendframework/zendxml               1.2.0              Utility library for XML usage, best practices, and security in PHP
zendframework/zftool                v0.1.0             Utility module for Zend Framework 2 applications.
zf-commons/zfc-base                 v0.1.2             A set of genetic (abstract) classes which are commonly used across multiple modules.
zf-commons/zfc-user                 1.4.4              A generic user registration and authentication module for ZF2. Supports Zend\Db and Doctrine2.
zf-commons/zfc-user-doctrine-orm    1.0.2              Doctrine2 ORM storage adapter for ZfcUser.
```

**Comment by jarrettj** [comment]
Had to manually install symfony/contracts. Surely that should be done automatically? 

Here's a whole bunch of new errors:
```
root@1d91ac0fae3d:/var/www/html# php vendor/bin/codecept run --env docker-local-chrome --fail-fast -vvvCodeception PHP Testing Framework v4.1.1
Powered by PHPUnit 5.7.27 by Sebastian Bergmann and contributors.
Running with seed: 


  [Connecting To Db] {"config":{"populate":true,"cleanup":false,"reconnect":false,"waitlock":0,"dump":"module/Application/tests/_data/dump.sql","populator":null,"dsn":"mysql:host=legal_testing_mysql;dbname=ci_test_legal","user":"root","password":"root"},"options":[]}
  [Db] Connected to default ci_test_legal
  Rebuilding WebGuy...

Acceptance (docker-local-chrome) Tests (77) ----------------------------------------------------------------------------------------------------------------------------------
Modules: WebDriver, Db, Redis
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CrudCest: Skip create. Actions are hardcoded.
Signature: Admin\Action\CrudCest:create
Test: module/Application/tests/acceptance/Admin/Action/CrudCest.php:create
Scenario --
 I login "lms-admin-user@mailinator.com","password"
   I load session snapshot "lms-admin-user@mailinator.com"
   I am on page "/user/login"
  [GET] https://legal_testing_app/user/login
  Unable to retrieve screenshot from Selenium : Session [(null externalkey)] not available and is not among the last 1000 terminated sessions.
  Active sessions are[ext. key 1382d97769b9b78a16862d387722de68]
  Unable to retrieve source page from Selenium : Session [(null externalkey)] not available and is not among the last 1000 terminated sessions.
  Active sessions are[ext. key 1382d97769b9b78a16862d387722de68]
  Screenshot and page source were saved into '/var/www/html/module/Application/tests/_log/' dir
 ERROR 

------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


Time: 20.73 seconds, Memory: 14.00MB

There was 1 error:

---------
1) CrudCest: Login as the admin user.
 Test  module/Application/tests/acceptance/Admin/Action/CrudCest.php:create
                                                                                                                                                                                                                       
  [Facebook\WebDriver\Exception\UnknownServerException] Session [(null externalkey)] not available and is not among the last 1000 terminated sessions.
Active sessions are[ext. key 1382d97769b9b78a16862d387722de68]  
                                                                                                                                                                                                                       

Scenario Steps:

 2. $I->amOnPage("/user/login") at module/Application/tests/_helpers/WebGuy.php:31
 1. $I->loadSessionSnapshot("lms-admin-user@mailinator.com") at module/Application/tests/_helpers/WebGuy.php:27

/var/www/html/vendor/facebook/webdriver/lib/Exception/WebDriverException.php:114
/var/www/html/vendor/facebook/webdriver/lib/Remote/HttpCommandExecutor.php:331
/var/www/html/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:565
/var/www/html/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:244
/var/www/html/vendor/codeception/module-webdriver/src/Codeception/Module/WebDriver.php:927
/var/www/html/vendor/codeception/codeception/src/Codeception/Step.php:266
/var/www/html/vendor/codeception/codeception/src/Codeception/Scenario.php:76
/var/www/html/module/Application/tests/_helpers/_generated/WebGuyActions.php:305
/var/www/html/module/Application/tests/_helpers/WebGuy.php:31
/var/www/html/module/Application/tests/acceptance/Admin/Action/CrudCest.php:14
/var/www/html/vendor/codeception/codeception/src/Codeception/Lib/Di.php:127
/var/www/html/vendor/codeception/codeception/src/Codeception/Test/Cest.php:138
/var/www/html/vendor/codeception/codeception/src/Codeception/Test/Cest.php:97
/var/www/html/vendor/codeception/codeception/src/Codeception/Test/Cest.php:80
/var/www/html/vendor/codeception/codeception/src/Codeception/Test/Test.php:88
/var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php:733
/var/www/html/vendor/codeception/phpunit-wrapper/src/Runner.php:110
/var/www/html/vendor/codeception/codeception/src/Codeception/SuiteManager.php:161
/var/www/html/vendor/codeception/codeception/src/Codeception/Codecept.php:196
/var/www/html/vendor/codeception/codeception/src/Codeception/Codecept.php:185
/var/www/html/vendor/codeception/codeception/src/Codeception/Command/Run.php:503
/var/www/html/vendor/codeception/codeception/src/Codeception/Command/Run.php:397
/var/www/html/vendor/symfony/console/Command/Command.php:255
/var/www/html/vendor/symfony/console/Application.php:982
/var/www/html/vendor/symfony/console/Application.php:255
/var/www/html/vendor/symfony/console/Application.php:148
/var/www/html/vendor/codeception/codeception/src/Codeception/Application.php:117
/var/www/html/vendor/codeception/codeception/app.php:46
/var/www/html/vendor/codeception/codeception/app.php:47
/var/www/html/vendor/codeception/codeception/codecept:7

Artifacts:

Png: /var/www/html/module/Application/tests/_log/Admin.Action.CrudCest.create.docker.local.chrome.fail.png
Html: /var/www/html/module/Application/tests/_log/Admin.Action.CrudCest.create.docker.local.chrome.fail.html

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
  [Db] Disconnected from default
  [Db] Disconnected from default
```

**Comment by jarrettj** [comment]
Seems the docker selenium was the issue. Downgraded a couple of versions and after lots of testing ended up on 3.141.59-uranium which seems to work again.

**Comment by Naktibalda** [comment]
Going back to your original error:

> PHP Fatal error:  Interface 'Symfony\Contracts\Service\ResetInterface' not found in /var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcherInterface.php on line 23

> symfony/event-dispatcher            v4.4.0             Symfony EventDispatcher Component
> symfony/event-dispatcher-contracts  v1.1.7             Generic abstractions related to dispatching event

> Had to manually install symfony/contracts. Surely that should be done automatically?

In Symfony v4 ResetInterface was part of [symfony/service-contracts](https://github.com/symfony/service-contracts/blob/master/ResetInterface.php) package.

For some reason it is only listed as `require-dev` dependency of symfony/event-dispatcher: https://github.com/symfony/event-dispatcher/blob/v4.4.0/composer.json#L27

Actually, the reason why it is in require-dev is that it was added to [Fix tests](https://github.com/symfony/event-dispatcher/commit/c71314cd3b9420b732e1526f33a24eff5430b5b3)

The only place where `TraceableEventDispatcherInterface` is mentioned is Codeception code which detects version of Symfony:

```php
        //TraceableEventDispatcherInterface was introduced in symfony/event-dispatcher 2.5 and removed in 5.0
        if (!interface_exists(TraceableEventDispatcherInterface::class)) {
            //Symfony 5
            $dispatcher->dispatch($eventObject, $eventType);
        } else {
            //Symfony 2,3 or 4
            $dispatcher->dispatch($eventType, $eventObject);
        }
```

https://github.com/Codeception/Codeception/blob/4.0/src/Codeception/Event/DispatcherWrapper.php

I will raise pull request to Symfony to fix dependencies.

**Comment by jarrettj** [comment]
Good day. Thanks @Naktibalda. Much appreciated.

**Comment by Naktibalda** [comment]
Alternatively, I could try to find another class that existed in EventDispatcher since at least Symfony 2.8 and disappeared in 5.0

**Comment by jarrettj** [comment]
I'm also thinking that there has to be another class yes. I can't be the only one with this issue. And maybe it is my environment.

**Comment by Naktibalda** [comment]
The reason why you are the first person to report this issue is that you have a mixture of Symfony 3 and 4 components:
```
symfony/browser-kit                 v4.4.0             Symfony BrowserKit Component
symfony/console                     v3.4.35            Symfony Console Component
symfony/css-selector                v4.4.0             Symfony CssSelector Component
symfony/debug                       v4.4.0             Symfony Debug Component
symfony/dom-crawler                 v4.4.0             Symfony DomCrawler Component
symfony/event-dispatcher            v4.4.0             Symfony EventDispatcher Component
symfony/event-dispatcher-contracts  v1.1.7             Generic abstractions related to dispatching event
symfony/filesystem                  v4.4.0             Symfony Filesystem Component
symfony/finder                      v4.4.0             Symfony Finder Component
symfony/polyfill-ctype              v1.12.0            Symfony polyfill for ctype functions
symfony/polyfill-mbstring           v1.12.0            Symfony polyfill for the Mbstring extension
symfony/process                     v3.4.35            Symfony Process Component
symfony/yaml                        v3.4.35            Symfony Yaml Component
```
Symfony/console v4.4 has dependency on `symfony/service-contracts` too, so if you upgraded it, this issue would have never happened.

**Comment by Naktibalda** [comment]
Detecting version of `symfony/event-dispatcher` is very error prone.

At first I checked if `Symfony\Contracts\EventDispatcher\Event` exists, but it quickly turned out that
`symfony/event-dispatcher-contracts` package can be installed with `symfony/event-dispatcher` v4.

I can't check existence of `Symfony\Component\EventDispatcher\Event` because Codeception creates a class alias of Contracts Event if it doesn't exists.

Current check has dependency on `symfony/console` matching major version of `symfony/event-dispatcher`.

Probably the best remaining option is to use Reflection to check if EventDispatcher has protected method doDispatch, which was removed in 5.0
https://github.com/symfony/event-dispatcher/blob/4.4/EventDispatcher.php#L258

There are more changes in Debug namespace, but they must be avoided because they can cause the same error as current solution.

**Comment by jarrettj** [comment]
To be honest, I can live with the composer warning of duplicate class names. As long as my test are running, that's all that matters :-D

**Comment by Naktibalda** [comment]
> To be honest, I can live with the composer warning of duplicate class names. 

If you upgraded `symfony/console`, remove `symfony/contracts`, you won't need it until upgrade to Symfony 5 components.

**Comment by Naktibalda** [comment]
EvenDispatcher version detection was fixed in 4.1.14 by #6066
